src/model/class.ts

```diff
@@ -81,6 +81,22 @@ export class ScannedClass implements ScannedFeature, Resolvable {
   resolve(document: Document): Feature|undefined {
     return new Class(this, document);
   }
+
+  finishInitialization(
+      methods: Map<string, ScannedMethod>,
+      properties: Map<string, ScannedProperty>) {
+    const mutableMethods = unsafeAsMutable(this.methods);
+    for (const [name, method] of methods) {
+      if (!mutableMethods.has(name)) {
+        mutableMethods.set(name, method);
+      }
+    }
+    for (const [name, prop] of properties) {
+      if (!this.properties.has(name)) {
```

Comment:
we currently skip methods/properties which already exist. i did this because something we picked up on the prototype is less reliable than an actual class method or property we already found i guess. though if we ever tried using this method elsewhere, that could be confusing.

Reply:
Hm, is it less reliable? Something on the prototype must have come after the class declaration, so it should override, yeah?

Reply:
that is very true, ill remove this check